runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7036,7 +7047,7 @@ TR::CompilationInfoPerThreadBase::preCompilationTasks(J9VMThread * vmThread,
                }
             }
 
-         if (!doLocalCompilation)
+         if (!doLocalCompilation && JITaaS::J9ClientStream::getVersionCheckStatus() != JITaaS::INCOMPATIBLE)
```

Comment:
If logic is added to `shouldPerformLOcalComp()` to check for the version, then this code is not needed.